src/containers/shared/components/Streams.jsx

```diff
@@ -230,14 +228,22 @@ class Streams extends Component {
 
   connect() {
     const rippledUrl = this.context;
-    const rippledWsUrl = `wss://${rippledUrl}:${process.env.REACT_APP_RIPPLED_WS_PORT}`;
-    this.ws = new WebSocket(rippledUrl == null ? DEFAULT_WS_URL : rippledWsUrl);
+    this.startWs(rippledUrl ?? process.env.REACT_APP_RIPPLED_HOST);
+  }
+
+  startWs(rippledUrl, wss = true) {
+    const connection = wss ? 'wss' : 'ws';
+    const wsUrl = `${connection}://${rippledUrl}:${process.env.REACT_APP_RIPPLED_WS_PORT}`;
+    this.ws = new WebSocket(wsUrl);
     this.ws.last = Date.now();
     Log.info(`connecting...`);
 
     // handle error
-    this.ws.onclose = () => {
+    this.ws.onclose = e => {
       Log.warn(`ws closed`);
+      if (!e.wasClean && wss) {
```

Comment:
nit: I think naming `wss` to `useWss` would be a bit more clear so people don't accidentally think `wss` stands for a websocket url.